Framework/Geometry/inc/MantidGeometry/Instrument/ComponentInfo.h

```diff
@@ -69,6 +71,27 @@ class MANTID_GEOMETRY_DLL ComponentInfo {
 
   /// Private copy constructor. Do not make public.
   ComponentInfo(const ComponentInfo &other);
+  template <typename IteratorT>
```

Comment:
I couldn't tell why these are templated, I saw only one use below?

Reply:
It's a boost reverse iterator being used. But it's simply being incremented, wanted to hide full type details from future readers.